### PR TITLE
[auto submit]fix errors, pass tests

### PR DESCRIPTION
**Pull request by XilaiZhang**
Yep auto submit wasn't using lint packages. geez i thought this was going to be fast, turned out there were quite many lint errors to fix.

### CODE REVIEW

File: auto_submit/analysis_options.yaml [hunk 1]
```diff
@@ -1,30 +1,52 @@
...
+    - ".dart_tool/**"
+    - "**/*.g.dart"
+    - "**/*.pb.dart"
+    - "**/*.pbjson.dart"
+    - "**/*.pbenum.dart"
+    - "test/src/utilities/mocks.mocks.dart"
 
-# For more information about the core and recommended set of lints, see
-# https://dart.dev/go/core-lints
+include: package:flutter_lints/flutter.yaml
```
Comment (godofredoc) [inline auto_submit/analysis_options.yaml:45]: is the unawaited rule part of this lints?

**Review by godofredoc — APPROVED** [review]
Thanks for the cleanup!

**Comment by XilaiZhang** [comment]
my pleasure

**Comment by CaseyHillers** [comment]
In the future `dart fix` should be able to auto-migrate most of this code.

**Pull request merged**